metabase/metabase
# Add timeline events restoring

Epic: https://github.com/metabase/metabase/issues/20289
Demo: https://www.loom.com/share/4ec4135cc7c84273a5c63c743374bc21

Changes:
- Adds a modal that displays a list of archived events for a timeline.
- Adds a `queryKey` to all requests where an entity is loaded by id. When the `query` used to load an entity changes, the cached entity will be invalidated. This is needed to reload timelines when navigating between modals displaying archived and non-archived data.

How to test:
- Open a timeline with some events
- Use some event's menu to archive it
- Use the timeline's menu to go to archived timelines
- Use some event's menu to restore it
- Make sure you can navigate with browser buttons between the regular and archived timeline modals and events are displayed correctly
